packages/dev/loaders/src/glTF/glTFFileLoader.ts

```diff
@@ -651,6 +665,32 @@ export class GLTFFileLoader implements IDisposable, ISceneLoaderPluginAsync, ISc
         );
     }
 
+    /**
+     * @internal
+     */
+    _loadBinary(
+        scene: Scene,
+        data: ArrayBufferView,
+        onSuccess: (data: any, responseURL?: string) => void,
+        onError?: (request?: WebRequest, exception?: LoadFileError) => void,
+        fileName?: string,
+    ): void {
+        const rootUrl = "file:";
+        const arrayBuffer = data.buffer.slice(data.byteOffset, data.byteLength);

```

Comment:
This shouldn't be necessary here. Only the validator requires a copy because it runs in a worker. DataReader should be able to handle `ArrayBufferView`s directly. We can modify the `_validate` function to take an ArrayBufferView and copy/transfer to worker when appropriate.